runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -2937,19 +2937,339 @@ J9::Z::TreeEvaluator::anewArrayEvaluator(TR::Node * node, TR::CodeGenerator * cg
       return TR::TreeEvaluator::VMnewEvaluator(node, cg);
    }
 
+TR::Register *
+performHelperCall(TR::Node *node, TR::SymbolReference *helperSymRef, TR::ILOpCodes helperCallOpCode, bool spillFPRegs, TR::CodeGenerator *cg)
+   {
+   TR::ILOpCodes opCode = node->getOpCodeValue();
+   TR::Node::recreate(node, helperCallOpCode);
+   if (helperSymRef)
+      {
+      node->setSymbolReference(helperSymRef);
+      }
+   TR::Register *targetReg = TR::TreeEvaluator::performCall(node, false /* isIndirect */, cg);
+   TR::Node::recreate(node, opCode);
+   return targetReg;
+   }
+
+TR::Node *
+createOutlinedCallNode(TR::CodeGenerator *cg, TR::Node *callNode, TR::ILOpCodes callOp)
```

Comment:
I don't see a need for this function to exist. I see that x86 has the exact same copy as well. We shouldn't be duplicating code like this which seems it could be shared across codegens. However I don't think we should do that as this function serves no purpose to me. We should be able to just evaluate the call as we do before this change, without "recreating" the call node.

Reply:
So this routine is used to create a copy of a node that can be passed to the `directCallEvaluator`. One reason we don't just pass the same node in is because we don't want the reference counts of the children to get decremented. This would be incorrect since we don't actually see the node appear another time in the trees. I think we can manually increase the counts before passing in the same node. This could solve that problem. I don't know if that's a better solution. Let me know what you prefer, and if there's a better solution here. I copied the entire routine from x86 here because I thought there's perhaps an opportunity to refactor here if we decide to use this routine.

Reply:
This routine no longer exists and we instead don't decrement the reference count in the multianewarrayEvaluator (it's done in the call we make to directCallEvaluator as per your recommendation. So the problem I mentioned above is no longer an issue.